tests/DCPS/Compiler/anonymous_types/test.idl

```diff
@@ -0,0 +1,116 @@
+#include <tao/LongSeq.pidl>
+
+typedef short Short3Array[3];
+typedef short Short2x3Array[2][3];
+typedef double Double2x3x4Array[2][3][4];
+typedef string Str2x3x4x5Array[2][3][4][5];
+
+typedef Short3Array Short3Array_x2[2];
+
+struct StructOfMultiDimArray {
+  Str2x3x4x5Array str2x3x4x5arr;
+  Short3Array_x2 s3arr_x2arr;
+};
+
+typedef sequence<long> LongSequence;
+typedef sequence<long, 3> Long3Sequence;
+typedef string<100> BoundedString;
+
+typedef struct XYZ_ {
+  double x;
+  double y;
+  double z;
+} XYZ;
+
+typedef sequence<XYZ> XYZSequence;
+
+@topic
+struct TestStruct {
+  char  c;
+  float f;
+  string str;
+  XYZ position;
+
+  Short3Array s3arr;
+  Short2x3Array s2x3arr;
+  Double2x3x4Array d2x3x4arr;
+  StructOfMultiDimArray smda;
+
+  LongSequence lseq;
+  Long3Sequence l3seq;
+  BoundedString bstr;
+  XYZSequence xyzseq;
+
+  // prefix a_ denotes anonymous type fields
+
+  short a_s3arr[3];
+  short a_s3arr_2nd[3];
+  @key short a_s2x3arr[2][3];
+  double a_d2x3x4arr[2][3][4];
+
+  sequence<long> a_lseq;
+  sequence<long> a_lseq_2nd;
+  sequence<long, 3> a_l3seq;
+  sequence<long, 3> a_l3seq_2nd;
+  sequence<long, 5> a_l5seq;
+  string<200> a_bstr;
+  sequence<XYZ> a_xyzseq;
+
+  //test C++ keyword used as anonymous type field
+  //to see all escaping is done correctly
+  sequence<long, 3> _long; // _tao_seq_CORBA_Long__3 _cxx_long;
+  sequence<long, 3> break; // _tao_seq_CORBA_Long__3 _cxx_break;
+  sequence<long, 3> class; // _tao_seq_CORBA_Long__3 _cxx_class;
+};
+
+// ========== ========== ========== ========== ==========
+//from Relay.idl
```

Comment:
Why is this here?  It doesn't seem to use anonymous IDL types.  Is it needed to test other features or compatibility of the generated code?

Reply:
Yes, the Relay.idl caught most errors that the anonymous type support code messed up the non-anonymous type support code. If we want to further factor out the code for both anonymous and non-anonymous type support, we need to test at least this much of the original Relay.idl to make sure they both work correctly. 

I just added the following in module RtpsRelay to test anonymous fields in module and topic:
  // test anonymous fields in module and topic
  @topic
  struct AnonymousFields {
    @key GUID_t guid;
    @key short a_s2x3arr[2][3];
    sequence<long, 5> a_l5seq;
  };

Reply:
I think you should boil down the IDL causing the error you were seeing and try to replicated them using the generic test IDL.